/*
* Copyright (C) 2008 Frederik M.J. Vestre
*
* Redistribution and use in source and binary forms, with or without
* modification, are permitted provided that the following conditions are met:
*     * Redistributions of source code must retain the above copyright
*       notice, this list of conditions and the following disclaimer.
*     * Redistributions in binary form must reproduce the above copyright
*       notice, this list of conditions and the following disclaimer in the
*       documentation and/or other materials provided with the distribution.
*
* THIS SOFTWARE IS PROVIDED BY <copyright holder> ''AS IS'' AND ANY
* EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
* WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
* DISCLAIMED. IN NO EVENT SHALL <copyright holder> BE LIABLE FOR ANY
* DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
* (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
* LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
* ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
* (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
* SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
*/

package protorpc;

option java_package = "com.likbilen.protorpc";
option java_outer_classname = "MessageProto";
option optimize_for = LITE_RUNTIME;

enum Type {
  REQUEST					= 1;
  RESPONSE					= 2;
  RESPONSE_CANCEL			= 3;
  RESPONSE_FAILED			= 4;
  RESPONSE_NOT_IMPLEMENTED	= 5;
  DISCONNECT				= 6;
  DESCRIPTOR_REQUEST		= 7;
  DESCRIPTOR_RESPONSE		= 8;
  MESSAGE                   = 9;
};

enum FunctionID {
  GetRestaurants                            = 1;
  GetLastestCommentsOfRestaurant            = 2;
  GetLastestCommentsByUser                  = 3;
  GetCommentsOfUserSince                    = 4;
  GetCommentsOfRestaurantSince              = 5;
  UserLogin                                 = 6;
  GetUserInfo                               = 7;
  GetRelatedUsers                           = 8;
  AddCommentForRestaurant                   = 9;
  AddRestaurant                             = 10;
  UpdateUserInfo                            = 11;
  SetUserRelation                           = 12;
  Search                                    = 13;
  GetSubscribtionInfo                       = 14;
};

message Message {
  optional Type type=1;
  optional FunctionID method_id=2;
  optional uint32 id =3;
  optional string name =4;	
  optional bytes buffer = 5;
}
